spirv_cross.cpp

```diff
@@ -1626,6 +1626,19 @@ SPIRBlock::ContinueBlockType Compiler::continue_block_type(const SPIRBlock &bloc
 	}
 }
 
+void Compiler::fix_switch_branches(const SPIRBlock &block) const
+{
+	auto search = ir.load_type_width.find(block.condition);
+	if (search == ir.load_type_width.end())
+	{
+		SPIRV_CROSS_THROW("Use of undeclared variable on a switch statement.");
```

Comment:
This breaks if the condition is a constant or spec constant.

Reply:
On that note -- is there a way to test shaders that would throw an exception?

Reply:
No. To be conformant, I think we actually have to support 64-bit switch statements don't we? That's a bit outside the scope of this PR though, I can test a bit locally before accepting the PR.

Reply:
Hmm, at least from what I can see in the latest spec ([GLSL lang spec 4.60](https://www.khronos.org/registry/OpenGL/specs/gl/GLSLangSpec.4.60.pdf)), `int` and `uint` are explicitly defined as 32 bits types [1] and in the definition of the `switch` statement it says the condition must be a scalar integer and if it's not, it's implicitly casted [2].

[1] Section 4.1.3 in the spec.
[2] Section 6.2

Reply:
We'd have to rewrite the switch to if/else ladders in this scenario.